### PR TITLE
task(settings): Support v2 key stretches for signin

### PR DESCRIPTION
**Pull request by dschom**
## Because

- Want  to support using V2 key stretching for react sign in

## This pull request

- Adds support for using v2 key stretching on sign in
- Updates functional tests to make sure V2 signin is active for react

## Issue that this pull request solves

Closes: FXA-9074

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

There was not an existing container.test.ts for sign in. Is there a ticket somewhere for this? We do have quite a bit of test coverage due to the keyStertchV2 tests, which cover a bunch of key stretching use cases and interchangeability between v1 and v2 key stretching.

### CODE REVIEW

**Review by chenba — APPROVED** [review]
LGTM.  Got a couple clean-up related comments.

(In the grand scheme of things, I still struggle to understanding why we'd want to proxy through gql-api.)

**Comment by dschom** [comment]
@chenba Thanks for the review.  By the way, proxying through GQL for this use case definitely felt awkward when compared to simply using auth-client directly.

File: packages/fxa-settings/src/pages/Signin/interfaces.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import internal from 'stream';
```
Comment (pdehaan) [inline packages/fxa-settings/src/pages/Signin/interfaces.ts:5]: Is this used somewhere? I don't see it in the diff on L107-139.

File: packages/fxa-auth-server/test/scripts/fixtures/accounts.sql [hunk 2]
```diff
@@ -18,6 +18,9 @@ CREATE TABLE `accounts` (
   `ecosystemAnonId` text CHARACTER SET ascii COLLATE ascii_bin,
   `disabledAt` bigint(20) unsigned DEFAULT NULL,
   `metricsOptOutAt` bigint(20) unsigned DEFAULT NULL,
+  `clientSalt` varchar(128) COLLATE utf8_unicode_ci DEFAULT NULL,
```
Comment (vbudhram) [inline packages/fxa-auth-server/test/scripts/fixtures/accounts.sql:21]: I wonder if there is an easy why to keep this file in Sync :/

File: packages/fxa-settings/src/pages/Signin/container.tsx [hunk 3]
```diff
@@ -147,20 +167,191 @@ const SigninContainer = ({
...
+
   const beginSigninHandler: BeginSigninHandler = useCallback(
     async (email: string, password: string) => {
       // TODO in oauth ticket
       // const service = integration.getService();
       const options = {
         verificationMethod: VerificationMethods.EMAIL_OTP,
       };
+
+      function handleGqlError(error: any) {
```
Comment (vbudhram) [inline packages/fxa-settings/src/pages/Signin/container.tsx:194]: Not for this PR, but I've seen the `handleGqlError` in several different components, slightly different each time. It seems like something we could refactor more generally.

File: packages/fxa-settings/src/pages/Signin/container.tsx [hunk 4]
```diff
@@ -147,20 +167,191 @@ const SigninContainer = ({
...
+      }
+
       // TODO in oauth ticket
       //   // keys must be true to receive keyFetchToken for oAuth and syncDesktop
       //   keys: isOAuth || isSyncDesktopV3,
       //   service: service !== MozServices.Default ? service : undefined,
       // };
       try {
         const { authPW } = await getCredentials(email, password);
+        console.log('!!! authPW', authPW);
```
Comment (vbudhram) [inline packages/fxa-settings/src/pages/Signin/container.tsx:354]: Stray log

**Review by vbudhram — COMMENTED** [review]
@dschom Thanks!

**Pull request merged**